pkg/crc/preflight/preflight_checks_network_linux.go

```diff
@@ -275,6 +276,20 @@ func checkNetworkManagerIsRunning() error {
 }
 
 func checkSystemdResolvedIsRunning() error {
+	resolvFilePath := "/etc/resolv.conf"
+	// Check /ETC/RESOLV.CONF section in `man systemd-resolved`
+	systemdResolvedManageResolveFilePath := []string{"/run/systemd/resolve/stub-resolv.conf",
+		"/usr/lib/systemd/resolv.conf",
+		"/run/systemd/resolve/resolv.conf"}
+	// Check if /etc/resolv.conf is managed by systemd-resolved
+	rFilePath, err := filepath.EvalSymlinks(resolvFilePath)
```

Comment:
This also seems to be available as part of `resolvectl status`?

Reply:
@cfergeau  I don't find it as part of status command

```
$ resolvectl status
Global
         Protocols: LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported
  resolv.conf mode: stub

Link 3 (wlp0s20f3)
    Current Scopes: none
         Protocols: -DefaultRoute LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported

Link 4 (virbr0)
    Current Scopes: LLMNR/IPv4
         Protocols: -DefaultRoute LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported

Link 21 (enp0s20f0u5u1)
    Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
         Protocols: +DefaultRoute LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported
Current DNS Server: 192.168.0.1
       DNS Servers: 192.168.0.1

Link 48 (cni-podman0)
    Current Scopes: none
         Protocols: -DefaultRoute LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported

Link 57 (vnet1)
    Current Scopes: LLMNR/IPv6
         Protocols: -DefaultRoute LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported

Link 68 (crc)
    Current Scopes: none
         Protocols: -DefaultRoute LLMNR=resolve -mDNS -DNSOverTLS DNSSEC=no/unsupported
       DNS Servers: 192.168.130.11
        DNS Domain: ~testing

```

Reply:
Isn't this `  resolv.conf mode: stub` ?

Reply:
As per man page of `systemd-resolved` there are 4 mode but it doesn't specify what `stuf` mode is for on csb I can see `foreign` as mode which doesn't have any info on the manual. Also how to parse the mode I can see there is option for json output and it doesn't provide any json value :( . Do we have a handy way to detect it and have docs around the mode value?

```
/ETC/RESOLV.CONF
       Four modes of handling /etc/resolv.conf (see resolv.conf(5)) are supported:

       •   systemd-resolved maintains the /run/systemd/resolve/stub-resolv.conf file for compatibility with traditional Linux programs. This file lists the 127.0.0.53 DNS stub (see above) as the only DNS server. It also contains a list of search
           domains that are in use by systemd-resolved. The list of search domains is always kept up-to-date. Note that /run/systemd/resolve/stub-resolv.conf should not be used directly by applications, but only through a symlink from
           /etc/resolv.conf. This file may be symlinked from /etc/resolv.conf in order to connect all local clients that bypass local DNS APIs to systemd-resolved with correct search domains settings. This mode of operation is recommended.

       •   A static file /usr/lib/systemd/resolv.conf is provided that lists the 127.0.0.53 DNS stub (see above) as only DNS server. This file may be symlinked from /etc/resolv.conf in order to connect all local clients that bypass local DNS APIs
           to systemd-resolved. This file does not contain any search domains.

       •   systemd-resolved maintains the /run/systemd/resolve/resolv.conf file for compatibility with traditional Linux programs. This file may be symlinked from /etc/resolv.conf and is always kept up-to-date, containing information about all
           known DNS servers. Note the file format's limitations: it does not know a concept of per-interface DNS servers and hence only contains system-wide DNS server definitions. Note that /run/systemd/resolve/resolv.conf should not be used
           directly by applications, but only through a symlink from /etc/resolv.conf. If this mode of operation is used local clients that bypass any local DNS API will also bypass systemd-resolved and will talk directly to the known DNS servers.

       •   Alternatively, /etc/resolv.conf may be managed by other packages, in which case systemd-resolved will read it for DNS configuration data. In this mode of operation systemd-resolved is consumer rather than provider of this configuration
           file.

       Note that the selected mode of operation for this file is detected fully automatically, depending on whether /etc/resolv.conf is a symlink to /run/systemd/resolve/resolv.conf or lists 127.0.0.53 as DNS server.
```

Reply:
To me `stub` is when there's a symlink to `/run/systemd/resolve/stub-resolv.conf`, and `foreign` would be `Alternatively, /etc/resolv.conf may be managed by other packages, in which case systemd-resolved will read it for DNS configuration data. In this mode of operation systemd-resolved is consumer rather than provider of this configuration file.`

This was merely a comment/suggestion, I agree it's not necessarily easier to use this, and the symlink seems like a "stable" enough interface, as it's explicitly documented in this man page.